azurerm/resource_arm_subnet.go

```diff
@@ -66,6 +66,39 @@ func resourceArmSubnet() *schema.Resource {
 				Optional: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+
+			"delegation": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"service_delegation": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"service_name": {
+										Type:     schema.TypeString,
+										Required: true,
```

Comment:
since the documentation below implies there's only one possible value here can we add validation for this:

```suggestion
										Required: true,
										ValidateFunc: validate.StringInSlice([]string{
										  "Microsoft.ContainerInstance/containerGroups",
										}, false)
```

Reply:
Actually, `Microsoft.ContainerInstance/containerGroups` is just one of possible values, you can get all available delegations through Azure CLI:
https://docs.microsoft.com/en-us/cli/azure/network/vnet/subnet?view=azure-cli-latest#az-network-vnet-subnet-list-available-delegations